src/components/input/input.ts

```diff
@@ -265,8 +268,14 @@ export default class SlInput extends ShoelaceElement implements ShoelaceFormCont
   }
 
   private handleInput() {
-    this.value = this.input.value;
-    this.formControlController.updateValidity();
+    if (this.currency === "usd") {
+      const numberWithCommas = this.input.value.replace(/\D/g, "").replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+      this.input.value = numberWithCommas;
```

Comment:
Moving this logic into this method is better than having it inline, but we're still modifying the actual value of the input, which could break forms when we try to use it.